build.rs

```diff
@@ -1,11 +1,50 @@
 extern crate bindgen;
 
+use std::env;
 use std::path::PathBuf;
 use std::process::Command;
 
 const HEADER_PATH: &str = "include/uapi/linux/nitro_enclaves.h";
 const OUT_FILE: &str = "/driver_structs.rs";
 
+// Generate the sysroot path based on the architecture the program is built.
+// This is needed for the compiler to locate the architecture dependent
+// header files.
+fn get_gcc_sysroot() -> String {
+    // Use the `CC` environment variable if existing
+    // which should give us directly the compiler's name
+    let cc = env::var("CC");
+    if let Ok(cc) = cc {
+        let output = Command::new(cc).arg("--print-sysroot").output();
+        if let Ok(output) = output {
+            let output = String::from_utf8_lossy(&output.stdout);
```

Comment:
Could you refactor the code a bit, so that we don't duplicate this part.
Something like:
First detect the compiler then parse the output, something like you already did for the last two code paths.

Reply:
Done.